azkaban-web-server/src/main/java/azkaban/imagemgmt/services/ImageTypeServiceImpl.java

```diff
@@ -60,6 +62,34 @@ public ImageTypeServiceImpl(final ImageTypeDao imageTypeDao,
     this.converter = converter;
   }
 
+  @Override
+  public List<ImageTypeDTO> getAllImageTypesWithOwnerships() {
+    List<ImageType> imageTypes = this.imageTypeDao.getAllImageTypesWithOwnerships();
+    List<ImageTypeDTO> imageTypeDTOs =
+        imageTypes.stream().map(i -> this.converter.convertToApiResponseDTO(i))
+            .collect(Collectors.toList());
+    return imageTypeDTOs;
+  }
+
+  @Override
+  public ImageTypeDTO findImageTypeWithOwnershipsById(String id) {
+    ImageType imageType = this.imageTypeDao.getImageTypeWithOwnershipsById(id);
+    return this.converter.convertToApiResponseDTO(imageType);
+  }
+
+  @Override
+  public ImageTypeDTO findImageTypeWithOwnershipsByName(String imageTypeName)
+      throws ImageMgmtException {
+    Optional<ImageType> imageType =
+        this.imageTypeDao.getImageTypeWithOwnershipsByName(imageTypeName);
+    if (imageType.isPresent()) {
+      return this.converter.convertToApiResponseDTO(imageType.get());
+    } else {
+      log.info("No Image Type Exists for ImageType name " + imageTypeName);
+      throw new ImageMgmtException("No Image Type Exists for ImageType name " + imageTypeName);
+    }
```

Comment:
I think `throw new ImageMgmtException("No Image Type Exists for ImageType name " + imageTypeName);` will generate a Bad Request response instead of a Not Found one. Could you check?

Reply:
Thank you, I will change it.